web/src/components/TaxRegistrationModal.test.tsx

```diff
@@ -450,6 +450,66 @@ describe("<TaxRegistrationModal>", () => {
       });
     });
   });
+
+  it("pre-populates user data from profile data", () => {
+    const userData = generateUserData({
+      profileData: generateProfileData({
+        legalStructureId: "limited-liability-partnership",
+        businessPersona: "STARTING",
+        taxId: "123456789000",
+        ownershipTypeIds: ["woman-owned"],
+        existingEmployees: "3",
+      }),
+    });
+
+    renderComponent(userData);
+
+    const woman = LookupOwnershipTypeById("woman-owned").name;
+
+    expect((screen.getByLabelText("Tax id") as HTMLInputElement).value).toEqual("123-456-789/000");
+    expect(screen.queryByLabelText("Business name")).not.toBeInTheDocument();
+    expect(screen.queryByLabelText("Ownership")).toHaveTextContent(`${woman}`);
+    expect((screen.getByLabelText("Existing employees") as HTMLInputElement).value).toEqual("3");
+  });
+
+  it("displays businessName field when it is not in profileData", () => {
+    const userData = generateUserData({
+      profileData: generateProfileData({
+        businessName: "",
+        legalStructureId: "limited-liability-partnership",
+        businessPersona: "STARTING",
+        ownershipTypeIds: ["woman-owned"],
+        existingEmployees: "3",
+      }),
+    });
+
+    renderComponent(userData);
+
+    expect(screen.getByText(Config.profileDefaults.STARTING.businessName.header)).toBeInTheDocument();
+  });
+
+  it("does not close modal and fires validation for tax id less than 12 digits", () => {
+    const userData = generateUserData({
+      profileData: generateProfileData({
+        legalStructureId: "limited-liability-partnership",
+        businessPersona: "STARTING",
+        businessName: "NJ Services",
+        taxId: "",
+        ownershipTypeIds: ["woman-owned"],
+        existingEmployees: "3",
+      }),
+    });
+
+    renderComponent(userData);
+    fireEvent.click(screen.getByLabelText("Tax id"));
+    fireEvent.change(screen.getByLabelText("Tax id"), { target: { value: "123456" } });
+    fireEvent.blur(screen.getByLabelText("Tax id"));
+    fireEvent.click(screen.getByTestId("modal-button-primary"));
+    expect(modalOnClose).not.toHaveBeenCalled();
+    expect(
+      screen.getByText(Config.profileDefaults[getFlow(userData.profileData)].taxId.errorTextRequired)
+    ).toBeInTheDocument();
+  });
```

Comment:
did you write all these tests? for that oneline change? or are you backfilling missing testst?

Reply:
it looks like these tests are duplicated above... is there a difference here that i'm missing?

Reply:
I had to move the test around because of a rebase, they were pre-existing.